### ISSUE TITLE
Data Loader outputting a different batch_size after the last iteration of an Epoch.

### ISSUE THREAD

**Issue by shamanez** [akanimax/Variational_Discriminator_Bottleneck#6]
I have followed how you have coded the custom data loader. It is amazing and I actually used it to for different other codes. But I found one interesting fact.  Lest assume **data** is the output from a custom data loader.  So **len(iter(data))** gives the total number of batches inside the entire data set. Let's say I have 101 data examples and my batch size is 10. Here  **len(iter(data))** gives me 11 (I thinks it is a ceil operation).  So the 11th batch only samples 1 value.  Then if you have a dynamic operation like calculating the GP in WGAN loss you need the batch_size. In case if you fix the batch size you will get an error. But you have used nice tricks such as **limit** and getting the batch size dynamically [line 1](https://github.com/akanimax/Variational_Discriminator_Bottleneck/blob/master/source/vdb/Gan.py#L196), [line 2](https://github.com/akanimax/Variational_Discriminator_Bottleneck/blob/master/source/vdb/Gan.py#L202).  Please tell me whether my understanding is correct or wrong?

**Comment by akanimax** [comment]
Hi @shamanez,

Sorry for the late reply. I have been going through a lot of work these days. Thanks for the appreciation. I just try to make my code easy to run as well as easy to read. That's all. :smile:.

So what you are saying is absolutely correct. `len(iter(data))` will give you the total number of batches. In general case, I preserve the last batch which could possibly be smaller than the fixed batch size. I don't understand why WGAN-GP loss calculation will be a problem with a batch size of 1. You'll still have 1 real and 1 fake image whose mean you can easily calculate. Even batch-normalization layers work properly with a possible batch size of 1. The only problem might be if you are calculating the `std` over the batch. I simply just turn off the bias correction in the std calculation. 

But if you'd still like a solution which will ensure the same batch size for every iteration, just simple add another keyword argument `drop_last=True` where the dataloader is defined [here](https://github.com/akanimax/Variational_Discriminator_Bottleneck/blob/master/source/data_processing/DataLoader.py#L187). For more information, you could read the documentation [here](https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader).

Hope this helps. 

Again sorry for the late reply.

Best regards,
@akanimax
